man/kernel-install.xml

```diff
@@ -233,6 +234,9 @@
       entry directories, and if found the entry token is derived from that. If that is not successful,
       <varname>$KERNEL_INSTALL_MACHINE_ID</varname> is used as fallback.</para>
 
+      <para><varname>$KERNEL_INSTALL_ROOT</varname> is set for the plugins to the absolute path of the
```

Comment:
This approach means we can't use --root in mkosi until all kernel-install plugins have been updated to take $KERNEL_INSTALL_ROOT into account. It's also super hard to do correctly because of possible absolute symlinks which need to be chased.

I think we should implement this by doing pivot_root() into the --root directory when we execute a plugin so that all existing plugins will automatically work with --root without changes

Reply:
Sounds nice.

Reply:
hmm, but what's our story there? plugins should originate in the image or from the host?

Reply:
Plugins should definitely originate in the image, they're usually distro specific and sometimes even distro version specific so picking them up outside the image wouldn't make sense.

Reply:
Maybe, better to add another boolean option `--chroot` or `--sandbox` or so, and make it defaults to yes when `--root` or `--image` is set. WDYT?

Reply:
hmm, then bootctl will become its own little container manager. Not sure how i feel about that...

Reply:
If we don't run plugins from the image then adding --root doesn't make much sense in the first place (at least for the install and remove verbs).

We could document that we leave APIVFS setup and such to the caller of kernel-install. The only thing we'd do in kernel-install would be unsharing the mount namespace and pivot rooting to the root directory before executing each plugin.